### PR TITLE
Fix: server does not terminate the process

### PR DESCRIPTION
**Pull request by NIKEA-SOFT**
Server may not shutdown the process, it gets stuck in a loop, waiting for input to the console.

### CODE REVIEW

File: server/server.cpp [hunk 1]
```diff
@@ -354,8 +364,11 @@ server::~server()
     auto notifier =
         qobject_cast<detail::async_readline_wrapper *>(m_stdin_notifier);
     if (notifier) {
+      notifier->stop();
+      notifier->quit();
```
Comment (lmoureaux) [inline server/server.cpp:368]: `quit()` does nothing because the thread isn't running an event loop

File: server/server.h [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 // Qt
 #include <QObject>
+#include <atomic>
```
Comment (lmoureaux) [inline server/server.h:24]: I don't think `<atomic>` is part of Qt... can we put an empty line between them?

File: server/server.cpp (hunk 1 above)
Comment (lmoureaux) [reply server/server.cpp:368]: oh yes, my bad, it does run an event loop

File: server/server.h (hunk 2 above)
Comment (NIKEA-SOFT) [reply server/server.h:24]: > I don't think `<atomic>` is part of Qt... can we put an empty line between them?
Why, if std::atomic is used there, we do not need to guess whether atomic is included somewhere or not

Comment (lmoureaux) [reply server/server.h:24]: I'm simply suggesting this:
```C++
// Qt
#include <QObject>

#include <atomic>
```
So if in the future we include more of Qt or the STL, it doesn't become inconsistent.

Comment (NIKEA-SOFT) [reply server/server.h:25]: Done

**Comment by lmoureaux** [comment]
Requesting a test from @jwrober -- start a local game, end it, the server should be gone

**Review by jwrober — APPROVED** [review]

**Pull request merged**